go/vt/schema/online_ddl.go

```diff
@@ -151,18 +152,18 @@ func ReadTopo(ctx context.Context, conn topo.Conn, entryPath string) (*OnlineDDL
 	return onlineDDL, nil
 }
 
-// getOnlineDDLAction parses the given SQL into a statement and returns the action type of the DDL statement, or error
+// ParseOnlineDDLStatement parses the given SQL into a statement and returns the action type of the DDL statement, or error
 // if the statement is not a DDL
-func getOnlineDDLAction(sql string) (action sqlparser.DDLAction, ddlStmt sqlparser.DDLStatement, err error) {
+func ParseOnlineDDLStatement(sql string) (ddlStmt sqlparser.DDLStatement, action sqlparser.DDLAction, err error) {
 	stmt, err := sqlparser.Parse(sql)
 	if err != nil {
-		return action, ddlStmt, fmt.Errorf("Error parsing statement: SQL=%s, error=%+v", sql, err)
+		return ddlStmt, action, fmt.Errorf("Error parsing statement: SQL=%s, error=%+v", sql, err)
```

Comment:
:nit: :rant: This is confusing code. I know the old code did this, but we should fix it while we are in here. ddlStmt and action have not been assigned to here, so this is equivalent to `return nil, nil, fmt.Errorf("Error parsing statement: SQL=%s, error=%+v", sql, err)`

I would suggest to not use named return arguments unless they help readability or need to be changed by a defer.

Reply:
I personally find this to be more idiomatic, because I don't need to think about the actual datatype for those return variables and I don't need to think about what's the default return value -- it's implicitly done on my behalf. Also, to _me_, it more readable to see what's being returned semantically.

Having said that I don't feel strongly and I'll adapt per your review.

Reply:
resolved

Reply:
You are right to point out that confusing is highly subjective. 

Let me explain where I'm coming from. I've always felt that it's important to return empty or nil values from a function when an error occurs. Some functions allow you to return both a valid value _and_ an error (example: https://golang.org/src/io/io.go?s=11741:11795#L322), so I always make sure to explicitly return nil or empty values when the function doesn't have a valid value to return.

The named return argument pattern makes it a little more difficult for a casual reader to see if we are returning the result as far as we got when the error occurred, or an empty value to signal that nothing was actually done.

Reply:
The way I see it, when an error is returned, in vast majority of cases the other values do not matter (are expected to be ignored). Therefore, it is only a rare occasion where those values will be checked. To me the code is more readable where I understand what it is that I'm returning semantically (e.g. this is the `uuid` rather than `""`). And it's only the rare occasions where I need to worry about the actual values returned.

So I guess my confusion is the inversion of yours...